source/pythonConsole.py

```diff
@@ -335,6 +358,55 @@ def onInputChar(self, evt):
 			self.Close()
 			return
 		evt.Skip()
+	
+	def onInputPaste(self, evt):
+		# Test if text content is available
+		if not wx.TheClipboard.IsSupported(wx.DataFormat(wx.DF_TEXT)):
+			evt.Skip()
+			return
+		cpTextData = wx.TextDataObject()
+		if not wx.TheClipboard.Open():
+			evt.Skip()
+			return
+		try:
+			res = wx.TheClipboard.GetData(cpTextData)

```

Comment:
While I think your use of the wx clipboard code makes sense, I think I prefer using api.getClipData, which is much simpler and neither requires opening/closing the clipboard nor checking the type of the data on the clipboard.

Reply:
Your are right, `api.getClipData` is at least much more concise here.
Unlike `wx.TheClipboard.GetData`, it does not automagically converts line-breaks, thus the use of `str.splitlines`.
Much cleaner anyways.